### DISCUSSION TITLE
v0.4.1

### DISCUSSION THREAD

**Discussion by Totodore** [Totodore/socketioxide#76] [Announcements]
# New version v0.4.1 🚀:

## This new release introduce : 
* Better error handling thanks to @biryukovmaxim 
* Engine.io v3 support thanks to @sleeyax and @biryukovmaxim 
* Streaming payload parsing to handle heavy payloads without buffering
* Max payload limit applied on input payload
* Specific transport list in configuration for socket.io and engine.io
* New [warp](https://github.com/seanmonstar/warp) integration example

## What's Changed
* chore: versioning by @Totodore in https://github.com/Totodore/socketioxide/pull/20
* Create CODE_OF_CONDUCT.md by @Totodore in https://github.com/Totodore/socketioxide/pull/23
* connection error handling: unsupported version, unsupported transport, bad handshake method by @biryukovmaxim in https://github.com/Totodore/socketioxide/pull/21
* Add Warp usage example & split examples and e2e testing by @Totodore in https://github.com/Totodore/socketioxide/pull/25
* Set content-type and charset by @biryukovmaxim in https://github.com/Totodore/socketioxide/pull/26
* handling send to socket error by @biryukovmaxim in https://github.com/Totodore/socketioxide/pull/31
* Revert "handling send to socket error" by @Totodore in https://github.com/Totodore/socketioxide/pull/33
* Handle send error by @biryukovmaxim in https://github.com/Totodore/socketioxide/pull/34
* Removing arc from handler property in `Engine` by @Totodore in https://github.com/Totodore/socketioxide/pull/32
* Create dependabot.yml by @Totodore in https://github.com/Totodore/socketioxide/pull/38
* Update itertools requirement from 0.10.5 to 0.11.0 in /socketioxide by @dependabot in https://github.com/Totodore/socketioxide/pull/39
* Feature: Error Adapter by @Totodore in https://github.com/Totodore/socketioxide/pull/40
* Set up code formatting by @Totodore in https://github.com/Totodore/socketioxide/pull/43
* Feature: implement EngineIO V3 (closes #22) by @sleeyax in https://github.com/Totodore/socketioxide/pull/24
* Fix failing EngineIO CI configuration by @sleeyax in https://github.com/Totodore/socketioxide/pull/47
* Add missing await by @sleeyax in https://github.com/Totodore/socketioxide/pull/48
* Fix invalid ping timeout in v3 protocol by @sleeyax in https://github.com/Totodore/socketioxide/pull/50
* fix(engineioxide/engine)-avoid-one-socket-cloning by @Totodore in https://github.com/Totodore/socketioxide/pull/57
* docs: add contributing guidelines by @Totodore in https://github.com/Totodore/socketioxide/pull/58
* Ft improve payload parsing by @biryukovmaxim in https://github.com/Totodore/socketioxide/pull/55
* Handle ordering bin packets by @biryukovmaxim in https://github.com/Totodore/socketioxide/pull/35
* Feat: improve payload parsing by @Totodore in https://github.com/Totodore/socketioxide/pull/53
* feat(engineioxide/v3) binary encoding payload by @Totodore in https://github.com/Totodore/socketioxide/pull/61
* chore(ci): add engineio ci for v4 and ci for v3 & v4 by @Totodore in https://github.com/Totodore/socketioxide/pull/56
* chore(deps): Update tokio-tungstenite requirement from 0.19.0 to 0.20.0 in /engineioxide by @dependabot in https://github.com/Totodore/socketioxide/pull/59
* feat(engineioxide): specific transport restrictions (fix issue #49) by @Totodore in https://github.com/Totodore/socketioxide/pull/70
* Create FUNDING.yml by @Totodore in https://github.com/Totodore/socketioxide/pull/72
* chore(versioning): v0.4.0 by @Totodore in https://github.com/Totodore/socketioxide/pull/73

## New Contributors
* @sleeyax made their first contribution in https://github.com/Totodore/socketioxide/pull/24

**Full Changelog**: https://github.com/Totodore/socketioxide/compare/v0.3.0...v0.4.1

<hr /><em>This discussion was created from the release <a href='https://github.com/Totodore/socketioxide/releases/tag/v0.4.1'>v0.4.1</a>.</em>

**Comment by tausifcreates** [comment]
Reddited the news https://www.reddit.com/r/rust/comments/15wcedr/rust_socketio_server_socketioxide_version_04/

**Comment by biryukovmaxim** [comment]
What about the twitter?

**Reply by Totodore** [reply]
I don't use Twitter so I didn't post any news on it. But feel free to post an announcement or something !
